app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -244,19 +248,80 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 	 * not authorized then the the [QueueMetadata] is created from the app remote API.
 	 */
 	fun createLikedSongsQueueMetadata() {
+		if (createQueueMetadataJob?.isActive == true) {
+			createQueueMetadataJob?.cancel()
+		}
+
 		createQueueMetadataJob = GlobalScope.launch(defaultDispatcher) {
 			queueItems = webApi.getLikedSongs(this@SpotifyAppController) ?: emptyList()
+
 			if (queueItems.isNotEmpty()) {
-				queueMetadata = QueueMetadata("Liked Songs", null, queueItems)
-				loadQueueCoverart()
+				queueMetadata = QueueMetadata(L.MUSIC_LIKED_SONGS_PLAYLIST_NAME, null, queueItems)
+
+				val hashCode = queueItems.hashCode().toString()
+				val likedSongsStateJson = appSettings[AppSettings.KEYS.SPOTIFY_LIKED_SONGS_PLAYLIST_STATE]
+				val likedSongsState: LikedSongsState
+				if (likedSongsStateJson.isBlank()) {
```

Comment:
This will get blanked out when the user Clears Storage. Can the WebApi offer a way to discover a previously-created AndroidAutoIdrive playlist instead of always creating? Maybe: !public, has the same queueTitle as the real `your_library` playlist, but is not the `your_library` playlist.

Reply:
I don't want to rely on string matching if possible because the translations are mutable.

Reply:
Ah that's a good idea to have a discovery of an already existing dummy playlist before creating a new one. Unfortunately it seems as though the only possible way to find the appropriate playlist will have to be through string matching either the name but as I mentioned in this comment here:  https://github.com/hufman/AndroidAutoIdrive/pull/263#discussion_r622723818 the name of the Liked Songs dummy playlist (SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME) will be a constant that should not be translated or tampered with so we can use this playlist title to find the appropriate playlist. The problem with the queueTitle approach you suggested is that the queueTitle will be translated and thus needs to be stored as to figure out what the dummy playlist is but then since the storage is being cleared we lose that and now we don't know what to look for. 

I added a new conditional check that if the likedSongsStateJson is blank then it will first check the user's playlists to see if there is an already existing dummy Liked Songs playlist. If that playlist is found it will pull the playlist uri and use that instead of creating a new one

Reply:
Ah, I had assumed the `SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME` was more visible, and thus would prefer to be translated. By having the translation being just the description, that does allow the name to be used as an identifier.